### PR TITLE
Improve stash api

### PR DESCRIPTION
**Pull request by jamal-khey**
DELETE /network-modifications (onlyStashed=false) -> delete all modifications (also stashed modification)
DELETE /network-modifications?onlyStashed=true -> delete all stashed modifications
GET /network-modifications (onlyStashed=false) -> get all modifications (also stashed modifications)
GET /network-modifications?onlyStashed=true -> get all stashed modifications
PUT /network-modifications?stashed=true -> stash modification
PUT /network-modifications?stashed=false -> restore modification

### CODE REVIEW

**Comment by SlimaneAmar** [comment]
Remove code smells

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 1]
```diff
@@ -1033,7 +1033,7 @@ public ResponseEntity<List<String>> getAvailableSvgComponentLibraries() {
     public ResponseEntity<String> getNetworkModifications(@Parameter(description = "Study UUID") @PathVariable("studyUuid") UUID studyUuid,
                                                           @Parameter(description = "Node UUID") @PathVariable("nodeUuid") UUID nodeUuid,
                                                           @Parameter(description = "Stashed Modification")
-                                                           @RequestParam(name = "stashed", required = false, defaultValue = "false") Boolean stashed) {
+                                                           @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean stashed) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1036]: Rename the argument ?
`onlyStashed`

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 2]
```diff
@@ -1068,34 +1068,35 @@ public ResponseEntity<Void> updateNetworkModification(@Parameter(description = "
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The network modifications was deleted"), @ApiResponse(responseCode = "404", description = "The study/node is not found")})
     public ResponseEntity<Void> deleteNetworkModifications(@Parameter(description = "Study UUID") @PathVariable("studyUuid") UUID studyUuid,
                                                            @Parameter(description = "Node UUID") @PathVariable("nodeUuid") UUID nodeUuid,
-                                                           @Parameter(description = "Network modification UUIDs") @RequestParam("uuids") List<UUID> networkModificationUuids,
+                                                           @Parameter(description = "Network modification UUIDs") @RequestParam(name = "uuids", required = false) List<UUID> networkModificationUuids,
+                                                           @Parameter(description = "Stashed Modification")
+                                                               @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean onlyStashed,
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1073]: Bad indentation

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 3]
```diff
@@ -1033,7 +1033,7 @@ public ResponseEntity<List<String>> getAvailableSvgComponentLibraries() {
     public ResponseEntity<String> getNetworkModifications(@Parameter(description = "Study UUID") @PathVariable("studyUuid") UUID studyUuid,
                                                           @Parameter(description = "Node UUID") @PathVariable("nodeUuid") UUID nodeUuid,
                                                           @Parameter(description = "Stashed Modification")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1035]: Change the description please

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 4]
```diff
@@ -1068,34 +1068,35 @@ public ResponseEntity<Void> updateNetworkModification(@Parameter(description = "
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The network modifications was deleted"), @ApiResponse(responseCode = "404", description = "The study/node is not found")})
     public ResponseEntity<Void> deleteNetworkModifications(@Parameter(description = "Study UUID") @PathVariable("studyUuid") UUID studyUuid,
                                                            @Parameter(description = "Node UUID") @PathVariable("nodeUuid") UUID nodeUuid,
-                                                           @Parameter(description = "Network modification UUIDs") @RequestParam("uuids") List<UUID> networkModificationUuids,
+                                                           @Parameter(description = "Network modification UUIDs") @RequestParam(name = "uuids", required = false) List<UUID> networkModificationUuids,
+                                                           @Parameter(description = "Stashed Modification")
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1072]: Change the description

File: src/main/java/org/gridsuite/study/server/StudyController.java [hunk 5]
```diff
@@ -1068,34 +1068,35 @@ public ResponseEntity<Void> updateNetworkModification(@Parameter(description = "
...
+                                                               @RequestParam(name = "onlyStashed", required = false, defaultValue = "false") Boolean onlyStashed,
                                                            @RequestHeader(HEADER_USER_ID) String userId) {
         studyService.assertCanModifyNode(studyUuid, nodeUuid);
-        studyService.deleteNetworkModifications(studyUuid, nodeUuid, networkModificationUuids, userId);
+        if (networkModificationUuids == null) {
+            studyService.deleteAllNetworkModifications(studyUuid, nodeUuid, onlyStashed, userId);
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyController.java:1077]: Use only one method ?

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 4 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 1 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 2 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

File: src/main/java/org/gridsuite/study/server/StudyController.java (hunk 5 above)
Comment (jamal-khey) [reply src/main/java/org/gridsuite/study/server/StudyController.java]: done

**Review by etiennehomer — APPROVED** [review]

**Pull request merged**